todo_list_app/utils.py

```diff
@@ -1,9 +1,81 @@
 # utils.py
+import pytz
+import redis
+import random
+from datetime import datetime
+from todo_list_app.models import Task
 from django.core.mail import send_mail
 from django.conf import settings
+from todo_list import settings
+from django.core.mail import EmailMessage
+from celery import shared_task
 
 
 def send_reminder_email(reminder):
     subject = f'Reminder: {reminder.task.title}'
     message = f'Hi {reminder.user.username},\n\nThis is a reminder for your task: {reminder.task.title}.'
-    send_mail(subject, message, settings.DEFAULT_FROM_EMAIL, [reminder.user.email])
\ No newline at end of file
+    send_mail(subject, message, settings.DEFAULT_FROM_EMAIL, [reminder.user.email])
+
+
+def local_to_utc(user_timezone, reminder):
+    if user_timezone:
+        user_tz = pytz.timezone(user_timezone)
+    else:
+        user_tz = pytz.utc
+    remind_at = datetime.strptime(reminder.remind_at, '%Y-%m-%dT%H:%M')
+    localized_remind_at = user_tz.localize(remind_at, is_dst=None)
+    reminder.remind_at = localized_remind_at.astimezone(pytz.utc)
+
+
+def reminder_create_or_update(user, reminder, timezone, task_id):
+    local_to_utc(timezone, reminder)
+    reminder.user = user
+    reminder.task = Task.objects.get(id=task_id)
+    return reminder.save()
+
+
+def generate_otp():
+    otp = random.randint(100000, 999999)
+    return otp
+
+
+@shared_task
+def send_email_task(subject, message, from_email, recipient_list):
+    send_mail(subject, message, from_email, recipient_list)
+
+
+def send_code_to_user(email, user_id):
+    redis_client = redis.StrictRedis.from_url(settings.CACHES['default']['LOCATION'])
+    subject = "One Time Passcode for Email Verification"
+    otp_code = generate_otp()
+    key = f"ev-{user_id}"
+    redis_client.set(key, str(otp_code), ex=120)
+    current_site = "TodoList.com"
+    email_body = f"Hi thanks for signing up on {current_site}. Please verify your email with the \n one time passcode {otp_code}"
+    from_email = settings.DEFAULT_FROM_EMAIL
+    send_email_task.delay(subject, email_body, from_email, [email])
```

Comment:
How can we run a function in celery after five minutes once?

Reply:
We can use apply_async instead of delay with countdown parameter and set it to 300 seconds
something like this
send_email_task.apply_async(args = [subject, email_body, from_email, [email]], countdown = 300)